SecondStage/module28/src/test/java/page/CommonAbstractPage.java

```diff
@@ -0,0 +1,34 @@
+package page;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+
+abstract public class CommonAbstractPage {
+
+    protected final int WAIT_TIMEOUT_SECONDS = 30;
+    protected WebDriver driver;
+    protected Logger logger = LogManager.getLogger();
+    protected static String keeperOfCalculatedCost;
+
+    public CommonAbstractPage(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    protected static ExpectedCondition<Boolean> jQueryAJAXsCompleted() {
```

Comment:
This solution is interesting and progressive. But in this case, it overcomplicates the code and causes tests to fail. I'd suggest to simply it a bit by verifying whether the page is completely loaded by checking the presence of some specific elements or page loaded condition may be using JS executor. Eg 
`public void waitForPageLoadIsComplete() {
        ExpectedCondition<Boolean> pageLoadFinishedCondition = driver -> ((JavascriptExecutor) Objects.requireNonNull(driver)).executeScript(
                "return document.readyState").equals("complete");
    }`